### ISSUE TITLE
Idris reads sources out of path set by env var $TARGET when set.

### ISSUE THREAD

**Issue by LeifW** [idris-lang/Idris-dev#4074]
# Steps to Reproduce
* cd into a git clone of an Idris library (e.g. https://github.com/ziman/lightyear or https://github.com/edwinb/SDL-idris)
* run `mkdir /tmp/foo; TARGET=/tmp/foo idris --install lightyear.ipkg`

# Expected Behavior
The .ibc files are copied into a directory within /tmp/foo

# Observed Behavior
None of the imports can be found:
```
TARGET=/tmp/idrisdir idris --build lightyear.ipkg
Can't find import Builtins
Can't find import Prelude
Can't find import Control/Monad/Tran
```
for lightyear, and
```
TARGET=/tmp/idrisdir idris --install sdl.ipkg
gcc: error: /tmp/idrisdir/rts/libtest.c: No such file or directory
Uncaught error: user error (Missing library SDL)
```
for SDL-idirs

This functionality has been used to make system packages out of these libraries, e.g. https://aur.archlinux.org/packages/idris-lightyear-git/ and https://aur.archlinux.org/packages/idris-sdl-git/

They first run `idris --build foo.ipkg` to compile the .ibcs, then run `TARGET=$(idris --libdir) idris --install foo.ipkg` to copy those .ibcs into the the appropriate dir for idris to read from. The `install` part is run as an unprivileged user, and its are zipped up into a tarball to be distributed and installed by a user with write permissions to `/usr/share/idris/libs`

**Comment by ahmadsalim** [comment]
@LeifW Thanks for reporting the issue.
I think there is an `--ibcdir` command, is that not what you are looking for?

I will close issue, unless I misunderstood the important of reading from `TARGET`.

**Comment by LeifW** [comment]
The `--ibcsubdir` argument works in place of setting `TARGET` for setting the install target dir, thanks.
I saw that was changed in https://github.com/idris-lang/Idris-dev/commit/b949b2846#diff-c561fe13665c2e2c16ab2ff62b99c596R38

I'm curious how the new `TARGET` env var is used. The old use I thought was *nix standard to set the install target for the "make; make install" process. Maybe it's used here? https://github.com/idris-lang/Idris-dev/blob/master/Setup.hs#L299
Having `TARGET` be the source of the libraries seems unintuitive to me. Is this feature actually used? Perhaps it could be used, if not. But if it works...
